x/poolmanager/routes.go

```diff
@@ -0,0 +1,537 @@
+	for _, route := range directPoolIDs {
+		pool, err := k.GetPool(ctx, route[0].PoolId)
+		if err != nil {
+			return nil, err
+		}
+		poolDenoms := pool.GetPoolDenoms(ctx)
+		liqInOsmo := osmomath.ZeroInt()
+		for _, denom := range poolDenoms {
+			liquidity, err := k.getPoolLiquidityOfDenom(ctx, route[0].PoolId, denom)
+			if err != nil {
+				return nil, err
+			}
+			liqInOsmoInternal, err := k.inputAmountToOSMO(ctx, denom, liquidity, routeMap)
+			if err != nil {
+				return nil, err
+			}
+
+			if pool.GetType() == types.Concentrated {
+				liqInOsmoInternal = liqInOsmoInternal.ToLegacyDec().Mul(superfluidMultiplier).TruncateInt()
+			}
+
+			liqInOsmo = liqInOsmo.Add(liqInOsmoInternal)
+		}
+		routeKey := fmt.Sprintf("%v", route)
+		routeLiquidity[routeKey] = liqInOsmo
+	}
+
+	// Check liquidity for all two-hop routes
+	for _, routes := range twoHopPoolIDs {
+		totalLiquidityInOsmo := osmomath.ZeroInt()
+		routeKey := fmt.Sprintf("%v", routes)
+		for _, route := range routes {
+			pool, err := k.GetPool(ctx, route.PoolId)
+			if err != nil {
+				return nil, err
+			}
+			poolDenoms := pool.GetPoolDenoms(ctx)
+			liqInOsmo := osmomath.ZeroInt()
+			for _, denom := range poolDenoms {
+				liquidity, err := k.getPoolLiquidityOfDenom(ctx, route.PoolId, denom)
+				if err != nil {
+					return nil, err
+				}
+				liqInOsmoInternal, err := k.inputAmountToOSMO(ctx, denom, liquidity, routeMap)
+				if err != nil {
+					return nil, err
+				}
+
+				if pool.GetType() == types.Concentrated {
+					liqInOsmoInternal = liqInOsmoInternal.ToLegacyDec().Mul(superfluidMultiplier).TruncateInt()
+				}
+				liqInOsmo = liqInOsmo.Add(liqInOsmoInternal)
+			}
+
+			totalLiquidityInOsmo = totalLiquidityInOsmo.Add(liqInOsmo)
+		}
+		routeLiquidity[routeKey] = totalLiquidityInOsmo
+	}
+
+	// Check liquidity for all three-hop routes
+	for _, routes := range threeHopPoolIDs {
+		totalLiquidityInOsmo := osmomath.ZeroInt()
+		routeKey := fmt.Sprintf("%v", routes)
+		for _, route := range routes {
+			pool, err := k.GetPool(ctx, route.PoolId)
+			if err != nil {
+				return nil, err
+			}
+			poolDenoms := pool.GetPoolDenoms(ctx)
+			liqInOsmo := osmomath.ZeroInt()
+			for _, denom := range poolDenoms {
+				liquidity, err := k.getPoolLiquidityOfDenom(ctx, route.PoolId, denom)
+				if err != nil {
+					return nil, err
+				}
+				liqInOsmoInternal, err := k.inputAmountToOSMO(ctx, denom, liquidity, routeMap)
+				if err != nil {
+					return nil, err
+				}
+
+				if pool.GetType() == types.Concentrated {
+					liqInOsmoInternal = liqInOsmoInternal.ToLegacyDec().Mul(superfluidMultiplier).TruncateInt()
+				}
+				liqInOsmo = liqInOsmo.Add(liqInOsmoInternal)
+			}
+			totalLiquidityInOsmo = totalLiquidityInOsmo.Add(liqInOsmo)
+		}
+		routeLiquidity[routeKey] = totalLiquidityInOsmo
+	}
+
+	// Extract and sort the keys from the routeLiquidity map
+	// This ensures deterministic selection of the best route
+	var keys []string
+	for k := range routeLiquidity {
```

Comment:
Just curious, why not use routeliquidity Map directly sorted instead of creating a new string arr?